pkg/disttask/framework/taskexecutor/task_executor.go

```diff
@@ -680,15 +683,24 @@ func (s *BaseTaskExecutor) markSubTaskCanceledOrFailed(ctx context.Context, subt
 }
 
 func (s *BaseTaskExecutor) updateErrorToSubtask(ctx context.Context, taskID int64, err error) error {
+	err = errors.Cause(err)
 	logger := logutil.Logger(s.logCtx)
-	backoffer := backoff.NewExponential(scheduler.RetrySQLInterval, 2, scheduler.RetrySQLMaxInterval)
-	err1 := handle.RunWithRetry(s.logCtx, scheduler.RetrySQLTimes, backoffer, logger,
-		func(_ context.Context) (bool, error) {
-			return true, s.taskTable.UpdateErrorToSubtask(ctx, s.id, taskID, err)
-		},
-	)
-	if err1 == nil {
-		logger.Warn("update error to subtask success", zap.Error(err))
+	if s.IsRetryableError(err) {
+		logger.Warn("meet retryable error", zap.Error(err))
+	} else if common.IsContextCanceledError(err) {
```

Comment:
seems we still need the `ctx cause` handling as in `markSubTaskCanceledOrFailed`